src/core.rs

```diff
@@ -952,6 +968,13 @@ impl Core {
     }
 
     fn client_identify(&mut self, peer_id: PeerId) -> Result<(), RoutingError> {
+        if self.proxy_candidates.iter().any(|&(id, _)| id == peer_id) {
+            return Ok(());
```

Comment:
Shouldn't we warn here that we are attempting to send a `ClientIdentify` to the same proxy twice?